### ISSUE TITLE
Extend `Waiter.Wait()` with context parameter

### ISSUE THREAD

**Issue by AnnaShaleva** [nspcc-dev/neo-go#3773]
## Is your feature request related to a problem? Please describe.
Inspired by @mialbu. `Waiter.Wait()` doesn't accept context, although `Waiter.WaitAny()` does. If user need to pass some custom cancellation context to awaiter, then he need to use `Waiter.WaitAny()` just because of the fact that `Waiter.Wait()` doesn't accept extra contexts.

## Describe the solution you'd like
Extend `Waiter.Wait()` with context parameter.

## Describe alternatives you've considered
Keep it as is, it's a matter of discussion. Opinions are welcomed.

**Comment by roman-khimov** [comment]
The only problem is that this is a breaking change.

**Comment by AnnaShaleva** [comment]
Yes, I thought about it, but at the same time it's very easy to migrate from old to new API. And in the past we already included such similar breaking changes (like https://github.com/nspcc-dev/neo-go/pull/3661, although I agree that it's not as widely-used by external users comparing to Actor's functionality).
